src/box/raft.c

```diff
@@ -451,34 +456,38 @@ box_raft_wait_term_outcome_f(struct trigger *trig, void *event)
 }
 
 int
-box_raft_wait_term_outcome(void)
+box_raft_promote_try(void)
 {
 	struct raft *raft = box_raft();
+	assert(raft->is_enabled);
+	assert(box_election_mode == ELECTION_MODE_MANUAL ||
+	       box_election_mode == ELECTION_MODE_CANDIDATE);
+
+	raft_promote(raft);
+
 	struct trigger trig;
 	struct box_raft_watch_ctx ctx = {
 		.is_done = false,
 		.term = raft->volatile_term,
 		.owner = fiber(),
 	};
-	trigger_create(&trig, box_raft_wait_term_outcome_f, &ctx, NULL);
+	trigger_create(&trig, box_raft_promote_try_f, &ctx, NULL);
 	raft_on_update(raft, &trig);
-	/*
-	 * XXX: it is not a good idea not to have a timeout here. If all nodes
```

Comment:
Maybe we can. I didn't care about that here though. Sending nacks might be complicated though. It would mean a response to a specific notification, which means they would need to have some kind of request IDs. Don't see an obvious solution here.